### ISSUE TITLE
Sharing dialog: Names sorting is case sensitive 

### ISSUE THREAD

**Issue by jaller94** [nextcloud/server#1094]
### Steps to reproduce
1. Create users: "Peter A.", "peter B.", "Petra".
2. Create a folder and open the Sharing dialog.
3. Type "Pet" into the people search.

![owncloud-sharing-name-sorting](https://cloud.githubusercontent.com/assets/10872136/18022304/fa7d7634-6bed-11e6-8123-ccd2c94610b1.png)

### Expected behaviour
The names should be ordered by the alphabet and case-insensitive. They are sorted like that on the Admin page "Users". "Petra" should be after "peter B.".

### Actual behaviour
"Petra" gets displayed before "peter B." because he starts on a lower p.

### Server configuration
**Operating system**: Linux 2.6.32

**Web server:** Apache

**Database:** mariadb 10.0.25

**PHP version:** 7.0.8

**Nextcloud version:** 10.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from 9.0.53

**Where did you install Nextcloud from:** nextcloud.com

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

### Client configuration
**Browser:** Firefox 45 / Chrome 51

**Operating system:** Ubuntu 16.04 / Windows 7

**Comment by rullzer** [comment]
CC: @nickvergessen probabaly something we have to take care of in the sharee api

**Comment by nickvergessen** [comment]
Not really, the user backend needs to do the sorting, otherwise paging will get broken/unexpected behaviour.

**Comment by skjnldsv** [comment]
Fixed with https://github.com/nextcloud/server/pull/15719
